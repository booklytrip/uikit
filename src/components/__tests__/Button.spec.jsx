import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button', () => {
    it('renders without crashing', () => {
        mount(<Button />);
    });

    it('should call onClick event if clicked', () => {
        const spy = jest.fn();
        const button = mount(<Button onClick={spy} />);
        button.simulate('click');

        expect(spy).toHaveBeenCalled();
    });

    describe('Snapshots', () => {
        it('renders primary button correctly', () => {
            const rendered = renderer.create(<Button primary />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders basic button correctly', () => {
            const rendered = renderer.create(<Button basic />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders loading button correctly', () => {
            const rendered = renderer.create(<Button loading />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders disabled button correctly', () => {
            const rendered = renderer.create(<Button disabled />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
