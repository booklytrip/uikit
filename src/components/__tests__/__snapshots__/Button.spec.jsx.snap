exports[`Button Snapshots renders basic button correctly 1`] = `
<button
  className="ui button basic"
  onClick={[Function]}
  style={undefined}
  type={undefined} />
`;

exports[`Button Snapshots renders disabled button correctly 1`] = `
<button
  className="ui button disabled"
  onClick={[Function]}
  style={undefined}
  type={undefined} />
`;

exports[`Button Snapshots renders loading button correctly 1`] = `
<button
  className="ui button loading"
  onClick={[Function]}
  style={undefined}
  type={undefined} />
`;

exports[`Button Snapshots renders primary button correctly 1`] = `
<button
  className="ui button primary"
  onClick={[Function]}
  style={undefined}
  type={undefined} />
`;
